modules/core/src/passes/draw-layers-pass.js

```diff
@@ -33,4 +42,36 @@ export default class DrawLayersPass extends LayersPass {
     }
     return parameters;
   }
+
+  // Pre-project point light positions here
+  getEffectParameters(viewport, coordinateSystem, coordinateOrigin, effectProps) {
+    if (effectProps && effectProps.lightSources) {
+      const {pointLights} = effectProps.lightSources;
```

Comment:
Shouldn't this be part of the LightEffect class?

Reply:
no, point light's projection is tight to viewport and layer's coordinateSystem, effect and effectManager doesn't know this information. In the layers-pass where we have these viewport and layer information we shouldn't see the effect objects, only the final props(uniform, texture, etc.) from effects. In long term once we have a unified world space we won't need this projection code.